pkg/server/event_handler.go

```diff
@@ -331,7 +332,7 @@ func (h *EventHandler[T, P]) getGCSObjectProto(ctx context.Context, objAttrs map
 	// Convert the object yaml bytes into a proto message wrapper.
 	p := P(new(T))
```

Comment:
I'm wondering maybe we should refactor the `getGCSObjectProto` method to just get gcs object bytes and we have another function to convert the object bytes to proto. Currently it is a little bit weird that the get gcs object will include pmaperrors. It's better that the function has just one single responsibility.

Reply:
Decoupled.